ligo/src/registryDAO.mligo

```diff
@@ -115,19 +127,38 @@ let apply_diff_affected (proposal_key, diff, ce : proposal_key *
 // Configuration Lambdas
 
 (*
- * Proposal check lambda : returns true if following two checks are true.
+ * Proposal check lambda : returns true if following checks are true.
  * 1. if proposer has locked frozen_scale_value * s + frozen_extra_value
  * where s = size(pack(proposalMetadata)) and frozen_scale_value and frozen_extra_value
  * are from storage configuration.
  * 2. proposal size < max_proposal_size where max_proposal_size is from stored configuration.
+ * 3. if transfers is present in the metadata, then it contains transfers such that each transfer:
+ *   3.a. is a xtz transfer satisfying min_xtz_amount <= amount <= max_xtz_amount
+ *   3.b. is a token transfer
+ * where min_xtz_amount and max_xtz_amount are from storage configuration.
  *)
 let registry_DAO_proposal_check (params, extras : propose_params * contract_extra) : bool =
   let proposal_size = Bytes.size(Bytes.pack(params.proposal_metadata)) in
   let frozen_scale_value = require_config ("frozen_scale_value", extras) in
   let frozen_extra_value = require_config ("frozen_extra_value", extras) in
   let max_proposal_size = require_config ("max_proposal_size", extras) in
-  let required_token_lock = frozen_scale_value * proposal_size + frozen_extra_value
-  in (params.frozen_token = required_token_lock) && (proposal_size < max_proposal_size)
+
+  let required_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
+  let has_correct_token_lock =
+    (params.frozen_token = required_token_lock) && (proposal_size < max_proposal_size) in
+
+  let ts = unpack_transfer_type_list_safe ("transfers", params.proposal_metadata) in
+  match ts with
+  | None -> has_correct_token_lock
+  | Some ts ->
+    let min_xtz_amount = unpack_tez ("min_xtz_amount", extras) in
+    let max_xtz_amount = unpack_tez ("max_xtz_amount", extras) in
+    let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
+      match transfer_type with
+      | Token_transfer_type tt -> is_valid
+      | Xtz_transfer_type xt -> is_valid && min_xtz_amount <= xt.amount && xt.amount <= max_xtz_amount
+    in
+    List.fold is_all_transfers_valid ts has_correct_token_lock
```

Comment:
Same comment as I made for treasuryDAO: no point on folding over the list (or even unpacking it, in this case) if `has_correct_token_lock` failed.